chain/cosmos/chain_node.go

```diff
@@ -321,6 +321,18 @@ func (tn *ChainNode) CreateKey(ctx context.Context, name string) error {
 	return dockerutil.HandleNodeJobError(tn.NodeJob(ctx, command))
 }
 
+// RecoverKey restores a key from a given mnemonic.
+func (tn *ChainNode) RecoverKey(ctx context.Context, keyName, mnemonic string) error {
+	command := []string{
+		tn.Chain.Config().GetShell(),
+		"-c",
+		fmt.Sprintf(`echo "%s" | %s keys add %s --recover --keyring-backend %s --home %s --output json`, mnemonic, tn.Chain.Config().Bin, keyName, keyring.BackendTest, tn.NodeHome()),
```

Comment:
`echo %q | ...` should be safer, as that will surround the string with quotes and add some escaping of special characters.

Reply:
Thanks for this, I wasn't familiar with %q here!